Sources/NIOHTTPCompression/HTTPResponseCompressor.swift

```diff
@@ -106,7 +106,7 @@ public final class HTTPResponseCompressor: ChannelDuplexHandler, RemovableChanne
         let httpData = unwrapOutboundIn(data)
         switch httpData {
         case .head(var responseHead):
-            guard let algorithm = compressionAlgorithm() else {
+            guard let algorithm = compressionAlgorithm(), responseHead.status != .noContent else {
```

Comment:
Should this check be generalized to see if the response has a body by checking content-length / transfer-encoding headers instead?

Reply:
No, checking for those headers is not a conclusive way to check if a response has a body. We _can_ check for it by using `HTTPResponseStatus.mayHaveResponseBody`: if the value of that is `false` we can safely skip this work.